### PR TITLE
[procutil] add OpenFdCount data collection

### PR DESCRIPTION
**Pull request by shang-wang**
### What does this PR do?

This PR adds collection for open fd count metric. The function body is pretty much reusing the code from gopsutil.
@DataDog/processes 

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

Write there any instructions and details you may have to test your PR.

### CODE REVIEW

**Review by robertjli — APPROVED** [review]

File: pkg/process/procutil/process_linux.go [hunk 1]
```diff
@@ -530,6 +531,27 @@ func (p *Probe) getLinkWithAuthCheck(pidPath string, file string) string {
 	return str
 }
 
+// getFDCount gets num_fds from /proc/(pid)/fd
+func (p *Probe) getFDCount(pidPath string) int32 {
+	path := filepath.Join(pidPath, "fd")
+
+	if err := p.ensurePathReadable(path); err != nil {
+		return 0
```
Comment (mbotarro) [inline pkg/process/procutil/process_linux.go:539]: Would it be useful to return `-1` when there's an error during the collection so we can differentiate absence of value and errors?

**Review by mbotarro — APPROVED** [review]

File: pkg/process/procutil/process_linux.go (hunk 1 above)
Comment (shang-wang) [reply pkg/process/procutil/process_linux.go]: I updated to return -1, thanks.

File: pkg/process/procutil/process_linux.go [hunk 2]
```diff
@@ -530,6 +531,27 @@ func (p *Probe) getLinkWithAuthCheck(pidPath string, file string) string {
 	return str
 }
 
+// getFDCount gets num_fds from /proc/(pid)/fd
+func (p *Probe) getFDCount(pidPath string) int32 {
+	path := filepath.Join(pidPath, "fd")
+
+	if err := p.ensurePathReadable(path); err != nil {
+		return -1
+	}
```
[the remark this answers is not in the document]
Comment (sunhay) [reply pkg/process/procutil/process_linux.go:540]: I think its very likely that if one of them fails, all of them fails :)

[the remark this answers is not in the document]
Comment (shang-wang) [reply pkg/process/procutil/process_linux.go:540]: This is actually my unstructured week's plan. Want to get things out first then do the optimization.

**Pull request merged**